### PR TITLE
combine some send_api functions + update in contracts

### PR DESCRIPTION
**Pull request by dorin-iancu**
- `self.send().direct()` now takes a nonce as argument. `direct_nft` removed.
- combined `esdt_local_mint` and `esdt_nft_add_quantity`
- combined `esdt_local_burn` and `esdt_nft_burn`

### CODE REVIEW

File: elrond-wasm/src/api/send_api.rs [hunk 1]
```diff
@@ -131,6 +106,46 @@ pub trait SendApi: ErrorApi + Clone + Sized {
...
+	/// Note that EGLD can NOT be transfered with this function.  
+	fn transfer_esdt_via_async_call(
+		&self,
+		to: &Address,
+		token: &TokenIdentifier,
+		nonce: u64,
+		amount: &Self::AmountType,
+		data: &[u8],
+	) -> ! {
+		if nonce == 0 {
```
Comment (sasurobert) [inline elrond-wasm/src/api/send_api.rs:122]: can't we combine EGLD transfer here as well ?

File: elrond-wasm/src/api/send_api.rs [hunk 2]
```diff
@@ -131,6 +106,46 @@ pub trait SendApi: ErrorApi + Clone + Sized {
 		)
 	}
 
+	/// Performs a simple ESDT/NFT transfer, but via async call.  
+	/// As with any async call, this immediately terminates the execution of the current call.  
+	/// So only use as the last call in your endpoint.  
+	/// If you want to perform multiple transfers, use `self.send().direct()` instead.  
+	/// Note that EGLD can NOT be transfered with this function.
```
Comment (sasurobert) [inline elrond-wasm/src/api/send_api.rs:113]: so right new we have direct and transfer_via_async_call combined - can we do the same for executeOnDestination contract with token transfer ?

File: elrond-wasm/src/api/send_api.rs (hunk 1 above)
Comment (dorin-iancu) [reply elrond-wasm/src/api/send_api.rs:122]: Can EGLD be transferred with AsyncCall? Keep in mind this function doesn't call any endpoint, it performs a simple transfer.

File: elrond-wasm/src/api/send_api.rs (hunk 2 above)
Comment (dorin-iancu) [reply elrond-wasm/src/api/send_api.rs:113]: executeOnDest context is a bit of a special case. There is a separate way of doing this: Andrei Marinica created a way of importing a contract's proxy, which lets the "parent" contract be able to call endpoints as if they were simple functions. This proxy automatically handles #[payment] arguments and performs the needed contract call.

TL;DR: These functions are not used directly.

**Review by catalinnnn — APPROVED** [review]

**Review by catalinnnn — APPROVED** [review]
